### DISCUSSION TITLE
how to set pinia as a global store or how to set $store as pinia and not vueX

### DISCUSSION THREAD

**Discussion by ezur1** [vuejs/pinia#1497] [Help and Questions]
Hi,
so im trying to set pinia accessible in all pages and components and get rid of importing pinia store very time.
looked everywhere but i couldn't find a solution.
please help.
thanks!

**Comment by posva** [comment]
https://github.com/vuejs/pinia/discussions/1060

**Reply by ezur1** [reply]
thanks for the replay!
im using nuxt 2.. do i need to install this plugin?
it says nuxt already has it built in...
also can you help with an example of use?
thanks!

**Reply by posva** [reply]
Nuxt has an autoimport feature out of the box (https://v3.nuxtjs.org/guide/concepts/auto-imports)
But I couldn't find the part in the docs. IIRC, there is an option to add autoImports. Autocompletion should be helpful to find the path here!

**Comment by nuvmo** [comment]
The approach I've taken is to expose the store in a global property in `main.js` after the app is mounted:
```
app.use(pinia).mount('#app')
app.config.globalProperties.$store = useMainStore()
```
I'd be interested to know if there are any drawbacks to this?
